### PR TITLE
Add gzip flag to jaeger-query

### LINKED ISSUE #1899: Add flag allow disable gzip encoding on jaeger-query
### Problem
Some HTTP client cannot process gzip encoding automatically

I can't change the HTTP client behavior

### Proposal 

Add a flag, allow disable gzip encoding

```
--query.gzip    enable gzip compress (default true)
```

### PR DESCRIPTION
**Pull request by Sasasu**
Signed-off-by: lizhaolong (Sasasu) <su@sasasu.me>

## Which problem is this PR solving?
- Resolves #1899

## Short description of the changes
- Add a flag to jaeger-query to control gzip is enabled or not

### CODE REVIEW

File: cmd/query/app/flags.go [hunk 1]
```diff
@@ -52,7 +55,7 @@ func AddFlags(flagSet *flag.FlagSet) {
 	flagSet.String(queryStaticFiles, "", "The directory path override for the static assets for the UI")
 	flagSet.String(queryUIConfig, "", "The path to the UI configuration file in JSON format")
 	flagSet.Bool(queryTokenPropagation, false, "Allow propagation of bearer token to be used by storage plugins")
-
+	flagSet.Bool(queryGzip, true, "enable gzip compress")
```
Comment (pavolloffay) [inline cmd/query/app/flags.go]: * remove the empty line above
* Start the message with a capital letter  - `Gzip compression`

Comment (Sasasu) [reply cmd/query/app/flags.go]: resolved :+1:

**Comment by yurishkuro** [comment]
I am not sure we should be implementing these hacks to support a faulty client that indicates that understand compression via the `Accept-Encoding` header, but then is unable to handle it.

**Comment by Sasasu** [comment]
the client is `ribbon` with `eureka` with `RestTemplate(org.springframework.web.client.RestTemplate)` reported by another Java developer.

**Pull request closed without merging**